modules/oxide-components/src/test/ts/browser/components/Draggable.spec.tsx

```diff
@@ -197,4 +198,31 @@ describe('browser.draggable.Draggable', () => {
       );
     }).toThrow('Draggable compound components must be rendered within the Draggable component');
   });
+
+  it('TINY-13102: Should allow button inside Draggable.Handle to be clicked', async () => {
+    let clickCount = 0;
+    const handleClick = () => {
+      clickCount++;
+    };
+
+    const TestElementWithButton = (
+      <Draggable.Root>
+        <div data-testid={draggableTestId} style={{ width: 250, height: 500, backgroundColor: 'gray' }}>
+          <Draggable.Handle>
+            <div data-testid={draggableHandleTestId} style={{ width: '100%', height: 50, backgroundColor: 'black' }}>
+              <button data-testid="button-in-handle" onClick={handleClick}>Click me</button>
+            </div>
+          </Draggable.Handle>
+        </div>
+      </Draggable.Root>
+    );
+
+    const { getByTestId } = render(TestElementWithButton, { wrapper: Wrapper });
+    const button = getByTestId('button-in-handle').element() as HTMLButtonElement;
```

Comment:
I wonder if we should abstract stuff like this away to avoid the magic cast I guess technically it would still be a magic cast but not using the `foo as bar` pattern.